tck/src/main/java/org/reactivestreams/tck/PublisherVerification.java

```diff
@@ -884,8 +884,7 @@ public void required_spec309_requestNegativeNumberMustSignalIllegalArgumentExcep
       @Override
       public void run(Publisher<T> pub) throws Throwable {
         final ManualSubscriber<T> sub = env.newManualSubscriber(pub);
-        final Random r = new Random();
-        sub.request(-r.nextInt(Integer.MAX_VALUE));
+        sub.request(-1);
```

Comment:
0 is explicitly tested with `required_spec309_requestZeroMustSignalIllegalArgumentException`.

Restored randomness and subtacted 1.